README.md

```diff
@@ -1,9 +1,243 @@
+    future = executor.create_task(worker)  # dispatch worker to executor
+    wait_for_future(future)  # block until worker is done
+
+@ros_process.main()
+def main() -> None:
+    ros_process.load(graph)  # or main.load
+    while True:
+        work(input())
+
+if __name__ == "__main__":
+    main()
+```
+
+#### Setting up command-line single node applications
+
+```python
+import argparse
+import logging
+import time
+
+from typing import Any
+
+from bdai_ros2_wrappers.futures import wait_for_future
+from bdai_ros2_wrappers.node import Node
+import bdai_ros2_wrappers.process as ros_process
+
+class Application(Node):
+
+    def __init__(self, robot_name: str, **kwargs: Any) -> None:
+        super().__init__("app_node", **kwargs)
+        self.robot_name = robot_name
+
+def cli() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("robot_name")
+    parser.add_argument("-v", "--verbose", action="store_true")
+    # define ROS 2 aware process arguments based on CLI arguments
+    parser.set_default(process_args=lambda args: dict(
+        prebaked=False, forward_logging=args.verbose))
+    return parser
+
+@ros_process.main(cli())
+def main(args: argparse.Namespace) -> None:
+    # set process-wide node for log forwarding to take effect
+    with ros_process.managed(Application, args.robot_name) as main.node:
+        if args.verbose:
+            root = logging.getLogger()
+            root.setLevel(logging.INFO)
+        logging.info("Application started!")
+        ros_process.spin()  # or main.spin
+
+if __name__ == "__main__":
+    main()
+```
+
+Note that process arguments are set via CLI. Also, note that a process-wide node is set explicitly (rather than implicitly). 
+The same can be done for process-wide executors. This allows fine grained control over ROS 2 configuration.
+
+#### Writing ROS 2 aware libraries
+
+You can use process-wide APIs to fetch ROS 2 defaults deep down call hierarchies, when passing nodes explicitly adds unnecessary clutter:
+
+```python
+import time
+
+import bdai_ros2_wrappers.process as ros_process
+import bdai_ros2_wrappers.scope as ros_scope
+from bdai_ros2_wrappers.node import Node
+
+class SomeFeature:
+
+    def __init__(self) -> None:
+        self.node = ros_scope.load(Node, "my_node")
+
+    def shutdown(self) -> None:
+        ros_scope.unload(self.node)  # or leave it behind until process ends
+
+@ros_process.main()
+def main() -> None:
+    feature = SomeFeature()
+    try:
+        time.sleep(10.0)
+    finally:
+        feature.shutdown()
+
+if __name__ == "__main__":
+    main()
+```
+
+Note the use of `ros_scope` instead of `ros_process`. Both offer the same APIs, but the former reaches out to the innermost scope. This allows the library to work in more complex, multi-threaded, multi-scope applications, and for testing cases.
+
+#### Writing ROS 2 aware tests
+
+To write tests, setup a global ROS 2 aware scope instead of a process.
```

Comment:
```suggestion
To write tests, set up a global ROS 2 aware scope instead of a process.
```

Reply:
Fixed in https://github.com/bdaiinstitute/ros_utilities/pull/32/commits/efb0db0d1ef57bdecc49aaff0dcbe9222defb1e3.